src/ImageSharp/Processing/Processors/CLoneingImageProcessor.cs

```diff
@@ -0,0 +1,166 @@
+﻿// <copyright file="CloneingImageProcessor.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.Processing
+{
+    using System;
+    using System.Threading.Tasks;
+
+    using ImageSharp.PixelFormats;
+    using SixLabors.Primitives;
+
+    /// <summary>
+    /// Allows the application of processors to images.
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    internal abstract class CloneingImageProcessor<TPixel> : IImageProcessor<TPixel>, ICloningImageProcessor<TPixel>
```

Comment:
Typo: `Cloneing...`

Can't we avoid code duplication by subclassing `ImageProcessor<TPixel>`?